test/eventsource_test.js

```diff
@@ -126,16 +126,24 @@ function writeEvents (chunks) {
   }
 }
 
+function assertRange(min, max, value) {
+  if (value < min || value > max) {
+    throw new Error('' + value + ' was not in range [' + min + ', ' + max + ']');
+  }
+}
+
 describe('Parser', function () {
   it('parses multibyte characters', function (done) {
     createServer(function (err, server) {
       if (err) return done(err)
 
       server.on('request', writeEvents(['id: 1\ndata: €豆腐\n\n']))
       var es = new EventSource(server.url)
+      es.onerror = function () {}
```

Comment:
You _must_ set an error handler for any EventEmitter that can emit an error event, otherwise you'll kill everything when an unhandled error happens. Since the tests were creating streams that they weren't cleaning up, but they were cleaning up the test _servers_, those streams would all get an EOF (which doesn't count as an error event), then try to reconnect one second later, then get an error because the server had gone away, and at that point you'd get a spurious failure in whatever other test happened to be running at that moment. The reason we weren't seeing this is that running the entire test suite usually took less than one second.

Reply:
Adding all these `es.close` lines is obviously very clunky, but in callback-style async code it's hard to avoid doing that stuff manually. If we migrate the tests to async/await style, it becomes _much_ easier to implement a simple mechanism for creating a thing, running a block, and being guaranteed to clean up the thing; you can see many examples of this in the `launchdarkly/js-sdk-common` and `js-client-sdk` tests. So I would favor doing that.

However, that's a considerable change in the test environment and will entirely replace the current test code, at which point this is barely a fork any more in the usual sense— that is, it wouldn't be feasible for the upstream maintainer to ever accept our changes if we chose to submit them upstream. So we may want to consider detaching this from the upstream repo if that's possible to do.